File: src/scoring/scoreFlush.ts
import { Card } from "../interfaces/card.interface";

/**
 * Scores points for a 4-card or 5-card flush.
 */
export const scoreFlush = (
  hand: Card[],
  commonCard?: Card,
  isCrib: boolean = false
): number => {
  const firstSuit = hand[0].suit;
  const handFlush = hand.every((card) => card.suit === firstSuit);

  if (handFlush && commonCard?.suit === firstSuit) {
    return 5;
  }
  if (handFlush && !isCrib) {
    return 4;
  }
  return 0;
};

File: crib_calc/scoring/score_flush.py
from crib_calc.card import Card


def score_flush(
    hand: list[Card],
    common_card: Card | None = None,
    is_crib: bool = False,
) -> int:
    first_suit = hand[0].suit
    hand_flush = all(c.suit == first_suit for c in hand)

    if hand_flush and common_card is not None and common_card.suit == first_suit:
        return 5
    if hand_flush and not is_crib:
        return 4
    return 0
